tasks/Task.cpp

```diff
@@ -0,0 +1,238 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "Task.hpp"
+
+using namespace usbl_evologics;
+
+Task::Task(std::string const& name)
+    : TaskBase(name)
+{
+}
+
+Task::Task(std::string const& name, RTT::ExecutionEngine* engine)
+    : TaskBase(name, engine)
+{
+}
+
+Task::~Task()
+{
+}
+
+
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See Task.hpp for more detailed
+// documentation about them.
+
+bool Task::configureHook()
+{
+    // Creating the driver object
+    driver.reset(new Driver());
+    if (!_io_port.get().empty())
+        driver->openURI(_io_port.get());
+    setDriver(driver.get());
+
+    if (! TaskBase::configureHook())
+        return false;
+
+    DeviceSettings settings = _device_settings.get();
+    DeviceSettings current_settings = driver->getCurrentSetting();
+    if(_change_parameters.get())
+    {
+        //TODO need validation
+        updateDeviceParameters(settings, current_settings);
+    }
+
+    VersionNumbers device_info = driver->getFirmwareInformation();
+    if(device_info.firmwareVersion.find("1.7") == std::string::npos)
+        RTT::log(RTT::Error) << "Component was developed for firmware version \"1.7\" and actual version is: \""<< device_info.firmwareVersion <<"\". Be aware of eventual incompatibility." << std::endl;
+    RTT::log(RTT::Info) << "USBL's firmware information: Firmware version: \""<< device_info.firmwareVersion <<"\"; Physical and Data-Link layer protocol: \""<<
+            device_info.accousticVersion <<"\"; Manufacturer: \""<< device_info.manufacturer << "\"" << std::endl;
+
+    return true;
+}
+bool Task::startHook()
+{
+    if (! TaskBase::startHook())
+        return false;
+    return true;
+}
+void Task::updateHook()
+{
+    TaskBase::updateHook();
+}
+void Task::errorHook()
+{
+    TaskBase::errorHook();
+}
+void Task::stopHook()
+{
+    TaskBase::stopHook();
+}
+void Task::cleanupHook()
+{
+    TaskBase::cleanupHook();
+}
+void Task::processIO()
+{
+
+    _device_settings.set(driver->getCurrentSetting());
+
+    _acoustic_channel.write(getAcousticChannelparameters());
```

Comment:
You are going to do this every time there is at least one byte to read, which can happen often. Add a status_period property (of time base/Time) and update it only every status_period seconds. The update should probably be done in the updateHook instead of here (processIO is really meant to process IO).